.github/workflows/ci.yml

```diff
@@ -63,15 +63,15 @@ jobs:
         path: |
           /usr/local/cargo/registry
           /usr/local/cargo/git
-        key: ${{ runner.os }}-test-pg${{ matrix.pgversion }}-cargo-1.12.0
-        restore-keys: ${{ runner.os }}-test-pg${{ matrix.pgversion }}-cargo-1.11.0
+        key: ${{ runner.os }}-test-pg${{ matrix.pgversion }}-cargo-${{ hashFiles('Cargo.lock', '.github/workflows/ci.yml') }}
```

Comment:
How about hashing extension/Cargo.toml instead?  That would keep the cache freshening at the same frequency we currently intend (release-based), since that's where we change the version number.

Our Cargo.lock changes far less often, so the cache would fall farther behind if we use it.  Same for ci.yml .

I'm going to propose getting rid of these entirely after I finish with the unified build container image.

Reply:
All the versions in `extension/Cargo.lock` are also in the lockfile. Since we use the Toolkit version as the Cargo version of the extension the lockfile will change whenever we change the version:
https://github.com/timescale/timescaledb-toolkit/blob/d5eeb7b90b4e01bb795d0855423f6854cbeb2f17/Cargo.lock#L2312-L2314

So by using the lockfile the cache key will change when we:
- update dependency versions
- update the Toolkit version
- update the CI workflow

Reply:
Right, of course it is!  Thanks! :)